src/test-utils.ts

```diff
@@ -0,0 +1,34 @@
+import { Server } from 'http';
+import axios, { AxiosInstance } from 'axios';
+import Koa from 'koa';
+
+export type UseSchemaClientOptions = {
+  service: Koa;
+};
+
+export type UseServiceClientContext = {
+  client: AxiosInstance;
+};
+
+export const useServiceClient = ({ service }: UseSchemaClientOptions) => {
```

Comment:
🎉 once this was in place did the tests reveal the double encoding then?

Reply:
Yes -- if you look at the "before" state of the 2nd commit in this PR, you'll see that the server receives a not-fully-decoded string.